### PR TITLE
iOS video player codegeneration

### PR DESCRIPTION
**Pull request by nathanagez**
*Issue #225*

*Description of changes:*

- Adds iOS video player generation (performs edit on `.pbxproj` file of your XCode projet)
- Install Cocoapods dependencies (MobileVLCKit)
- Adds bridging header
- Refactor video player templates tree structure
- Trims empty lines from template files

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: provider-utils/awscloudformation/utils/video-player-utils.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  try {
+    fs.readFileSync(`${projectRootPath}/Podfile`, { encoding: 'utf-8' });
+    let podFileData = await exec('pod', ['ipc', 'podfile-json', 'Podfile'], false);
+    podFileData = JSON.parse(podFileData);
+    if (isVLCKitInstalled(podFileData, getProjectConfig(context).projectName)) {
+      context.print.info('Podfile already contains MobileVLCKit');
+      await exec('pod', ['install'], true);
+    } else {
+      const lines = await readLastLines.read(`${projectRootPath}/Podfile`, 1);
```
Comment (wizage) [inline provider-utils/awscloudformation/utils/video-player-utils.js:102]: I am worried about this way to edit the Podfile.

My concern is more complicated projects can have potential issues def if they don't follow the typical order:

I.E. They might have multiple targets and the top target is the one we want to add to not the bottom one.

Might be better to use a combo of Regex and prompting the user to ask what target they want to add for the pods file.

React-native does this today with the Podfile: https://github.com/lorenzo-deepcode/react-native/tree/master/local-cli/link/pods

We can follow in their footsteps as a good way to look for the right place to write in the file.

File: provider-utils/awscloudformation/video-player-templates/video-player.ejs [hunk 2]
```diff
@@ -139,4 +139,58 @@ export default Ember.Component.extend({
     });
   },
 });
-<% } %>
\ No newline at end of file
+<% } -%>
+<% if (framework === 'ios') { -%>
```
Comment (wizage) [inline provider-utils/awscloudformation/video-player-templates/video-player.ejs:143]: Can we split the iOS/Android ones out of the JS one? Might just look cleaner

**Review by wizage — CHANGES_REQUESTED** [review]
Let me know your thoughts on the review below

File: provider-utils/awscloudformation/utils/video-player-utils.js (hunk 1 above)
Comment (nathanagez) [reply provider-utils/awscloudformation/utils/video-player-utils.js:102]: Yes you are right, I will refactor this part :)

File: provider-utils/awscloudformation/video-player-templates/video-player.ejs (hunk 2 above)
Comment (nathanagez) [reply provider-utils/awscloudformation/video-player-templates/video-player.ejs:143]: Make sense! I added a `video-player.ejs` file in each directory (web/ios)

**Comment by wizage** [comment]
Closing in favor of #241

**Pull request closed without merging**